tests/middleware/test_trace_logging.py

```diff
@@ -48,26 +48,23 @@
 }
 
 
+async def app(scope, receive, send):
+    assert scope["type"] == "http"
+    await send({"type": "http.response.start", "status": 204, "headers": []})
+    await send({"type": "http.response.body", "body": b"", "more_body": False})
+
+
 @pytest.mark.skipif(
     sys.platform.startswith("win") or platform.python_implementation() == "PyPy",
     reason="Skipping test on Windows and PyPy",
 )
 def test_trace_logging(capsys):
-    class App:
-        def __init__(self, scope):
-            if scope["type"] != "http":
-                raise Exception()
-
-        async def __call__(self, receive, send):
-            await send({"type": "http.response.start", "status": 204, "headers": []})
-            await send({"type": "http.response.body", "body": b"", "more_body": False})
-
     class CustomServer(Server):
```

Comment:
I found myself having to change the CustomServer in 4 or 5 files while working on a draft today, maybe a good refactor for another pr to keep things separated or here have no preference, what do you think?

Reply:
Not sure what you mean here exactly — the `CustomServer` is left untouched? Only the ASGI apps are modified and factored out. Did you mean that the individual ASGI apps were modified as well?